static/app/views/integrationOrganizationLink/index.tsx

```diff
@@ -136,22 +133,24 @@ export default class IntegrationOrganizationLink extends AsyncView<Props, State>
   onInstallWithInstallationId = (data: Integration) => {
     const {organization} = this.state;
     const orgId = organization && organization.slug;
-    this.props.router.push(
-      normalizeUrl(`/settings/${orgId}/integrations/${data.provider.key}/${data.id}/`)
+    window.location.assign(
+      `${organization?.links.organizationUrl || ''}/settings/${orgId}/integrations/${
+        data.provider.key
+      }/${data.id}/`
```

Comment:
You should have `normalizeUrl()` on the path segment of this URL as under custom domains you'll hit an extra redirect without it.